src/Http/Auth/RequestAuthenticatorMiddleware.php

```diff
@@ -141,6 +145,12 @@ private function authenticateToken(ServerRequestInterface $request): void
 
         $validator->verifySignature($this->token);
         $validator->validateClaims($this->token);
+        if ($this->token->getEmailAddress()) {
+            $this->userEmailAddressRepository::addUserEmail(
+                $this->token->getUserId(),
+                $this->token->getEmailAddress()
+            );
+        }
```

Comment:
I'm wondering if you really need to set the email address on the repository here using a static method. I think you could just get the `JsonWebToken::class` "service" from the container in the `UserEmailAddressRepository` service definition (in `UserServiceProvider`) and inject it in the repository that way (using the constructor or another method like this one).

I wasn't sure before because the `JsonWebToken::class` gets its value from the `RequestAuthenticatorMiddleware::getToken()` method, which will return `null` until the middleware is called by the router with a `Request`: https://github.com/cultuurnet/udb3-backend/blob/4728e8a920e4aadc3cbdcb7aebcf10ba5839ee48/app/Authentication/AuthServiceProvider.php#L131

However, this request handler for example also gets `JsonWebToken` injected directly in its constructor upon instantiation: https://github.com/cultuurnet/udb3-backend/blob/bca8bb38ada8aa07806652fc93a77f9873e1119d/app/User/UserServiceProvider.php#L52

The reason why it works is that we register our routes with just the class name of the request handler, for example: https://github.com/cultuurnet/udb3-backend/blob/64770269f8545d4e37ce541e0ad3bcb6c1ef94be/app/Http/PsrRouterServiceProvider.php#L465

Later, after the router has called all the middlewares (and thus there is a `JsonWebToken` in most cases), the router uses our custom `InvokableRequestHandlerContainer` which looks up the given class name in the service container and returns it. It's only at that point that the request handler *and all its dependencies* are instantiated.

So that means that the command bus, the permission voters, and their dependency on your new `UserEmailAddressRepository` are only instantiated after there is already a `JsonWebToken` "service" in the container (if a token was used, remember that there are some routes that do not need a token so it can always be `null` after all)